libmscore/repeatlist.cpp

```diff
@@ -516,21 +720,60 @@ Measure* RepeatList::findStartRepeat(Measure* m)
 //---------------------------------------------------------
 //   findStartFromRepeatCount
 //    @param startFrom starting measure for this repeat subsection
-//    @param sectionEndMeasure final measure of the section in which to look for possible endRepeats
 //    @return number of times playback passes the start repeat barline (not accounting for jumps)
 //---------------------------------------------------------
 
-int RepeatList::findStartFromRepeatCount(Measure * const startFrom, Measure * const sectionEndMeasure)
+int RepeatList::findStartFromRepeatCount(Measure * const startFrom) const
       {
       Measure * m = startFrom;
       int startFromRepeatCount = (m->repeatEnd())? (m->repeatCount()) : 1;
       m = m->nextMeasure();
-      while (m && m != sectionEndMeasure->nextMeasure() && !m->repeatStart()) {
+      while (m && !m->repeatStart()) {
             if (m->repeatEnd()) {
                   startFromRepeatCount += m->repeatCount() - 1;
                   }
-            m = m->nextMeasure();
+            m = (m->sectionBreak()) ? nullptr : m->nextMeasure();
             }
       return startFromRepeatCount;
       }
+
+//---------------------------------------------------------
+//   isFinalPlaythrough
+//    @param measure the measure to verify
+//    @param repeatSegment consider measure with its playbackCount in this specific RepeatSegment
+//    @return true if that measure has its final playthrough in the given repeatSegment
+//---------------------------------------------------------
+
+bool RepeatList::isFinalPlaythrough(Measure * const measure, QList<RepeatSegment*>::const_iterator repeatSegmentIt) const
```

Comment:
Why do we need const pointer to non-const data? What is the reason?

Reply:
The main reason is me writing my signature first, then my body.
Having the const-pointer will throw me a compilation error if I start writing stuff I thought I shouldn't be writing. So more of a declaration of intent.